holopy/scattering/calculations.py

```diff
@@ -124,10 +120,10 @@ def calc_holo(schema, scatterer, medium_index=None, illum_wavelen=None, illum_po
         Calculated hologram from the given distribution of spheres
     """
     theory = interpret_theory(scatterer,theory)
-    par = optical_parameters(schema, medium_index=medium_index, illum_wavelen=illum_wavelen, illum_polarization=illum_polarization)
-    scat = theory._calc_field(scatterer, schema, **par)
-    holo = scattered_field_to_hologram(scat*scaling, par['illum_polarization'], schema.normals)
-    return finalize(schema, holo)
+    uschema = update_metadata(schema, medium_index, illum_wavelen, illum_polarization)
```

Comment:
Why are you using update_metadata in some places and optical_parameters in others?